snap/snapcraft.yaml

```
@@ -0,0 +1,33 @@
+name: flamenco
+base: core22
+version: '1.0.0'
+summary: Package maintenance tool for Ubuntu Toolchains
+description: |
+  Flamenco is a CLI tool that helps toolchain developers manage many
+  different package versions and releases from a single debian
+  folder source tree.
+
+grade: stable
+confinement: strict
+
+architectures:
+  - build-on: [amd64]
+    build-for: [amd64]
```

suggestion: I would also like for this to build for arm64 (maybe also s390x and ppc64el in the future too)

```suggestion
  - build-on: [amd64]
    build-for: [amd64]
  - build-on: [amd64]
    build-for: [arm64]
```

---

Yes, that's what I had originally.

However, in the `parts` section, you'll see that we have `dotnet-self-contained-runtime-identifier: linux-x64`. This would need to be `dotnet-self-contained-runtime-identifier: linux-arm64` whenever we're building for `arm64`.

The plugin doesn't currently support variable input on that field, so we're stuck with this until we either add support for the content snap, whenever we have it, or tweak the plugin to support variable input on that field.

---

We could also do it manually with `plugin: nil`. This would also have the advantage that it uses the Ubuntu built SDK rather than Microsoft built SDK.

---

The dotnet plugin will use the Canonical SDK simply because of
https://github.com/canonical/flamenco/blob/d360539e2b7b37e313de016660b0a90f41307194/snap/snapcraft.yaml#L23-L24

I could write some bash magic to publish according to `CRAFT_ARCH_BUILD_FOR` and make the entire build process by hand, but the discussion falls into the trade-off of:
1. Making the arm64 build available right away and stop making flamenco a use-case for the dotnet plugin for now, or
2. Keep flamenco as a dotnet plugin use-case and publish the arm64 binary when the dotnet plugin supports that.

---

> The dotnet plugin will use the Ubuntu SDK simply because of
> 
> https://github.com/canonical/flamenco/blob/d360539e2b7b37e313de016660b0a90f41307194/snap/snapcraft.yaml#L23-L24

Forgot about that.

> I could write some bash magic to publish according to `CRAFT_ARCH_BUILD_FOR` and make the entire build process by hand, but the discussion falls into the trade-off of:
> 
>     1. Making the arm64 build available right away and stop making flamenco a use-case for the dotnet plugin for now, or
> 
>     2. Keep flamenco as a dotnet plugin use-case and publish the arm64 binary when the dotnet plugin supports that.

I prefer option 1. The current dotnet plugin has usability issues what this example demonstrates again and we have to change the `snapcraft.yaml` file either way (either to change back to the dotnet plugin or to adopt the improved dotnet plugin configuration properties). I am happy to switch to the dotnet plugin once it improves, but in the meantime we have more control over the build process.